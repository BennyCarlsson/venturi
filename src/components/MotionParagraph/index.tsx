import Paragraph, { ParagraphProps } from 'components/Paragraph'
import { HTMLMotionProps, motion } from 'framer-motion'

interface MotionParagraphProps extends HTMLMotionProps<'p'>, ParagraphProps {}

const MotionParagraph = ({ title, ..._props }: MotionParagraphProps) => {
  const props = { as: motion.p, ..._props }
  return <Paragraph {...props} />
}

export default MotionParagraph
