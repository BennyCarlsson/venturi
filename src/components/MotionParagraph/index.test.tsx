import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import MotionParagraph from ".";

describe("<MotionParagraph />", () => {
  it("should match snapshot", () => {
    const tree = renderer
      .create(<MotionParagraph>Texty text</MotionParagraph>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    render(<MotionParagraph>Texty text</MotionParagraph>);
    expect(screen).toBeTruthy();
  });

  it("should render text", () => {
    render(<MotionParagraph>Texty text</MotionParagraph>);
    expect(screen.getByText("Texty text")).toBeInTheDocument();
  });

  it("should render p tag", () => {
    render(<MotionParagraph>Texty text</MotionParagraph>);
    expect(
      screen.getByText("Texty text", { selector: "p" })
    ).toBeInTheDocument();
  });
});
