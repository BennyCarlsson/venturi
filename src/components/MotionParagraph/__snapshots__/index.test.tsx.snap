// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MotionParagraph /> should match snapshot 1`] = `
<p
  className="sc-bdfBwQ kMVXxV"
  style={Object {}}
>
  Texty text
</p>
`;
